terraform_utils/flatmap.go

```diff
@@ -137,7 +137,7 @@ func (p *FlatmapParser) fromFlatmapObject(prefix string, tys map[string]cty.Type
 		if !inAttributes {
 			continue
 		}
-		if p.isAttributeIgnored(attributeName) {
```

Comment:
What case/bug for this changes?

Reply:
There's a resource aws_cloudfront_distribution that have an attribute `domain_name` marked as ignored but there's also a nested attribute `origin.domain_name` [1] that is a required field. The code currently removes both attributes - top level one as well as nested ones, which breaks the HCL

[1] https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html#domain_name